FRENCH/src/translation-terms.md

```diff
@@ -73,48 +81,53 @@ Voici les principaux termes techniques qui ont été traduits de l'Anglais vers
 | PATH | PATH | - |
 | pattern | motif | - |
 | pattern-matching | filtrage par motif | - |
-| placeholder | espace réservé | - |
+| placeholder | espace réservé | `{}` pour `fmt` |
 | prelude | étape préliminaire | - |
-| procedural macros | macroprocédures | - |
-| project chapters | chapitre de projet | - |
-| raw identifiers | identificateurs primaires | - |
+| procedural macro | macro procédurale | - |
+| project chapter | chapitre de projet | - |
+| raw identifier | identificateur brut | - |
 | README | README | - |
 | refactoring | remaniement | - |
-| reference | réference | - |
+| reference | référence | - |
 | registry | registre | - |
 | regression | régression | - |
-| reproducible builds | compilation reproductive | - |
+| reproducible build | compilation reproductible | - |
 | run | exécuter | pour les programmes |
-| Rustaceans | Rustacés | - |
+| Rustacean | Rustacé | - |
 | section header | entête de section | - |
 | semantic version | version sémantique | - |
 | script | script | - |
-| shadow | créer un fantôme | explication variable type inference |
-| Shadowing | La création de fantôme | variable type inference |
-| shell | terminal/invite de commande | - |
+| shadow | masquer | remplacer une variable par une autre de même nom |

```

Comment:
I'm affraid that use of `masquer` instead of `créer un fantôme` in the context of ch02-00 (https://github.com/Jimskapt/rust-book-fr/commit/978c52d09944c63b2b62a4ff0317e5558f96d68c#diff-5ddef0abe825ee210fea3f0df46857e6R1621) does not help to understand what we want to say ?

Reply:
I've heard people say *masquer* as a translation of *to shadow* with regards to other programming languages.
In the context of ch02-00, *masquer* works fine:

> C'est le cas, mais Rust nous permet de masquer la précédente valeur de `deduction` par une nouvelle.

I'm not sure that the *fantôme* analogy is any clearer.

Reply:
Sorry, but I'm not really comfortable with the `masquer` term, because, if I understood well, the *original variable* still usable after *shadowing*. This french translation `masquer` really give me the impression that the new variable is replacing the origin.

That reminds me the french term [`surcharger`](https://fr.wikipedia.org/wiki/Surcharge_(programmation_informatique)), usually used for functions. Do you think that could be a good in-between ?

Reply:
Are you sure that the shadowed variable is still usable?
AFAIK, once a variable is shadowed, you can't access it directly.

Reply:
The `ch03-01` that i've translated today explains it more.

I think maybe you got right and the original variable is not again available, so in this case `masquer` will be appropriate ?

I need more time to think about it, and eventually experiment this. I'll think about it and I'll going back here soon.

Reply:
I've tried the folowing code :
https://gist.github.com/rust-play/e102f56d75d784e075923f857c8e203e

And it look like it is like you understood : the original variable is not available after *shadowing*.

In fact, the second `parse()` in the gist is done on the `i32` instead of the string, unlike the first `parse()` : so it does not compile.

So I think now we can use safely `masquer` to translate `shadowing`.